build_dev_binpkgs

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+# Copyright (c) 2023 by the Flatcar Maintainers.
+# Use of this source code is governed by the Apache 2.0 license.
+
+. "$(dirname "$0")/common.sh" || exit 1
+
+# Script must run inside the chroot
+assert_inside_chroot
+assert_not_root_user
+
+# Dependencies and packages to include by default.
+packages_default=( "coreos-devel/board-packages" )
+
+# Packages that are rdeps of the above but should not be included.
+# (mostly large packages, e.g. programming languages etc.)
+skip_packages_default="dev-lang/rust,virtual/rust,dev-lang/go,dev-lang/go-bootstrap,dev-go/go-md2man"
+
+
+# Developer-visible flags.
+DEFINE_string board "${DEFAULT_BOARD}" \
+  "The board to build packages for."
+DEFINE_string skip_packages "${skip_packages_default[@]}" \
+  "Comma-separated list of packages in the dependency tree to skip."
+DEFINE_boolean pretend "${FLAGS_FALSE}" \
+	"List packages that would be built but do not actually build."
+
+FLAGS_HELP="usage: $(basename $0) [flags] [packages]
+
+build_dev_binpkgs builds binary packages for all dependencies of [packages]
+that are not present in '/build/<board>/var/lib/portage/pkgs/'.
+Useful for publishing a complete set of packages to a binhost.
+
+[packages] defaults to '${packages_default}' if not specified.
+"
+
+# Parse command line
+FLAGS "$@" || exit 1
+eval set -- "${FLAGS_ARGV}"
+
+# Die on any errors.
+switch_to_strict_mode
+
+if [[ $# -eq 0 ]]; then
+  set -- "${packages_default[@]}"
+fi
+# --
+
+function my_board_emerge() {
+	export PORTAGE_CONFIGROOT="/build/${FLAGS_board}"
+	export SYSROOT="${SYSROOT:-/build/${FLAGS_board}}"
+	export ROOT="/build/${FLAGS_board}"
+	sudo -E emerge --root-deps=rdeps "${@}"
```

Comment:
The export is global, how about moving it to the invocation - this has the same effect for the command but does not leak to the function caller
```suggestion
	PORTAGE_CONFIGROOT="/build/${FLAGS_board}" SYSROOT="${SYSROOT:-/build/${FLAGS_board}}" ROOT="/build/${FLAGS_board}" sudo -E emerge --root-deps=rdeps "${@}"
```

Reply:
one can also do `sudo VAR=val cmds…` and drop the `-E` if it's only about these vars and not others